RFC/src/RFC-0310_AssetTemplates.md

```diff
@@ -0,0 +1,218 @@
+
+#### Asset identification
+
+Assets are identified by `<TLDI>.#_hhhhh` on the Tari network, where `#` is the template number, and `hhhhh`
+is a hash of the asset creation instruction.
+
+This allows assets to be deterministically identified from their initial state. Two different creation instructions
+leading to the same hash refer to the same single asset, by definition. Validator Nodes maintain an index of assets and
+their committees, and so can determine whether a given asset already exists; and MUST reject any `create_asset`
+instruction for an existing asset.
+
+#### Template number
+
+Tari uses templates to define the behaviour for its smart contracts. The template number refers to the type of digital
+asset begin created. The logic and rules associated with each template will be part of every[*] validator node's binary.
+
+[*]: # 'Except possibly for proprietary plugin templates running on permissioned committees'
+
+The template number is a 64-bit unsigned integer and has the following format, with 0 representing
+the least significant bit.
+
+| Bit range | Description                            |
+|:----------|:---------------------------------------|
+| 0 - 31    | Template number (0 - 4,294,967,295)    |
+| 32 - 47   | Template version (0 - 65,535)          |
+| 48 - 58   | Reserved  (Must be 0)                  |
+| 59        | Non-public nodes flag                  |
+| 60        | Confidentiality data flag              |
+| 61        | Proprietary contract flag              |
+| 62        | Public but non-community contract flag |
+| 63        | Beta Mode flag                         |
+
+The lowest 32 bits refer to the canonical smart contract type; the qualitative types of contracts the network supports.
+Many assets can be issued from a single template.
+
+Examples of template types may be
+
+| Template number | Asset                    |
+|:----------------|:-------------------------|
+| 1               | Simple single-use tokens |
+| 2               | Simple coupons           |
+| ...             | ...                      |
+| 120             | Collectible Cards        |
+| 144             | In-game items            |
+
+The template number may also set one or more feature flags to indicate that the contract is
+* experimental, or in testing phase, (bit 63).
+* a “corporate” template (i.e. it’s not a community-developed template, but one developed by an asset issuer to address
+  their specific needs - bit 62). However, the code for the contract is open-source and can be linked into validator
+  node binaries.
+* It’s possible that we allow “corporate” templates to be executed via plugins to the node software (i.e. they are not
+  part of the core Tari codebase at all). Thus these templates may be running proprietary code. (bit 61). It follows
+  that not all validator nodes will be aware of such contracts and that all assets running under such contracts would be
+  executed in [permissioned mode].
```

Comment:
see naming above. 

We need to decide on a name before this gets accepted so as not to confuse people afterwords

Reply:
Yeah, I'll apply those changes universally once we converge on a naming convention